File: src/class047/C1_CorporateFlightBookings.cpp
#include <vector>
using namespace std;

vector<int> corpFlightBookings(vector<vector<int>>& bookings, int n) {
    vector<int> cnt(n + 2, 0);
    for (auto& book : bookings) {
        cnt[book[0]] += book[2];
        cnt[book[1] + 1] -= book[2];
    }
    
    for (int i = 1; i < cnt.size(); ++i) {
        cnt[i] += cnt[i - 1];
    }
    
    return vector<int>(cnt.begin() + 1, cnt.begin() + n + 1);
}

File: src/class047/C1_CorporateFlightBookings.rs
fn corp_flight_bookings(bookings: Vec<Vec<i32>>, n: usize) -> Vec<i32> {
    let mut cnt = vec![0; n + 2];
    for book in &bookings {
        cnt[book[0] as usize] += book[2];
        cnt[book[1] as usize + 1] -= book[2];
    }
    
    for i in 1..cnt.len() {
        cnt[i] += cnt[i - 1];
    }
    
    cnt[1..=n].to_vec()
}
